jobbergate-docs/src/local-development.rst

```diff
@@ -0,0 +1,263 @@
+   ┡━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━┩
+   │ 6   │ test_chain             │ aaaaaaaa               │ bbbbbbb                            │ bdx@bdx.com                        │ False                │
+   │ 68  │ 00_smallcase_TEST_00_1 │ None                   │ smallcaseratssssss                 │ james.beedy@scania.com             │ True                 │
+   │ 67  │ 00_smallcase_TEST      │ None                   │ smallcaseratssssss                 │ james.beedy@scania.com             │ True                 │
+   │ 75  │ BDXTEST                │ None                   │ RATS                               │ james.beedy@scania.com             │ True                 │
+   │ 78  │ converge-test          │ converge-app           │                                    │ abdallah.el-hajjam@scania.com      │ True                 │
+   │ 84  │ 00_smallcase           │ None                   │ Small application for testing      │ james.beedy@scania.com             │ True                 │
+   │     │                        │                        │ jobbergate.                        │                                    │                      │
+   ...
+   │ 91  │ tucker-test            │ tucker-test            │                                    │ tucker@omnivector.solutions        │ True                 │
+   └─────┴────────────────────────┴────────────────────────┴────────────────────────────────────┴────────────────────────────────────┴──────────────────────┘
+
+If you see this kind of output, congratulations! You now have an local instance of
+``jobbergate-cli`` working and ready for develoment.
+
+Make sure to deactivate the Poetry virtualenv before you move on by typing "exit" or
+hitting ``<ctl-d>``.
+
+
+Local jobbergate-api setup
+--------------------------
+
+The ``jobbergate-api`` is easiest to set up for local development using
+``docker-compose``. There is a ``docker-compose.yml`` in the root directory of the main
+``jobbergate`` repository that can be used to spin up the following services:
+
+* The ``jobbergate-api``
+* A local ``postgres`` instance for the ``jobbergate-api`` to consume
+* A test ``postgres`` instance to use for the unit tests
+* A ``minio`` instance for mimicing file storage on ``S3``
+
+
+Create some basic config
+........................
+
+First, ``docker-compose`` will need a ``.env`` file that contains some basic settings.
+Create the ``.env`` file in the root ``jobbergate`` directory with the following
+contents::
+
+   AUTH0_DOMAIN=omnivector.us.auth0.com
+   AUTH0_AUDIENCE=https://armada.omnivector.solutions
+
+
+Stand it up with ``docker-compose``
+...................................
+
+If you haven't already, `Clone the git repo`_.  From the root directory of the
+``jobbergate`` directory, execute ``docker-compose``:
+
+.. code-block:: bash
+
+   $ docker-compose up --build
```

Comment:
The docker compose is not initializing the databases. To get the ready, the user should run:

.. code-block:: bash

    $ cd jobbergate-api
    $ poetry install
    $ poetry shell
    $ alembic --config=alembic/alembic.ini upgrade d3388a22e0e9
    $ alembic --config=alembic/alembic.ini upgrade 5f987ab31acf
    $ alembic --config=alembic/alembic.ini upgrade 75325d649ee2

Reply:
You need only run one command: `make db-upgrade`.
This will install dependencies (if they are not installed already) and issue the alembic commands.

Reply:
I added a TODO in the doc to add this. We will need some effort to figure out a way to do this automatically when the database is being started via docker-compose. See commit 80630b5